mrmustard/math/lattice/strategies/vanilla.py

```diff
@@ -72,7 +73,7 @@ def vanilla(shape: tuple[int, ...], A, b, c) -> ComplexTensor:  # pragma: no cov
             value_at_index += A[i, j] * np.sqrt(index_u[j]) * ret[n]
         ret[index] = value_at_index / np.sqrt(index_u[i])
 
-    return ret.reshape(shape)
+    return ret.reshape(shape) * c
```

Comment:
this leads to the same result, but this way we perform a lot more multiplications no? (one per entry of `ret`)

Reply:
I agree. But when c is no longer a scalar, but a polynomial we need to do an inner product between c and some of the dimensions of ret (what is returned from hermite renorm). This essentially updates c. Do you think one can incorporate that inner product in hermite renorm?

Reply:
Oh I see. But still, is it going work the way you wrote it? Because the `*` operator needs `ret` and `c` to have compatible shapes or one of them to be a scalar. is that the case?

Reply:
No. Right now we have to call hermite renorm with `c=1`, and then do the inner product afterwards. `ret` and `c` will typically have different dimensions and the inner product is performed between some of these dimensions. I will change it back to the way it was when c is just a scalar to reduce the number of multiplications. We can always look into maybe doing something smarter when c is a polynomial.